### PR TITLE
DOC: add examples for xlog1py

### PR DESCRIPTION
**Pull request by OmarManzoor**
#### Reference issue

Relates to gh-7168

#### What does this implement/fix?

- Adds an example for xlog1py

#### Additional information

None

### CODE REVIEW

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you for the PR @OmarManzoor. I have the same suggestions as your other PR. Some comments on the results and params you are taking would help.

File: scipy/special/_add_newdocs.py [hunk 1]
```diff
@@ -13103,6 +13103,43 @@ def add_newdoc(name, doc):
...
+    This example shows how the function can be used to calculate the log of
+    the probability mass function for a geometric discrete random variable.
+    The probability mass function of the geometric distribution is defined
+    as follows:
+
+    .. math:: f(k) = (1-p)^{k-1} p
+
+    where :math:`p` is the probability of a single success
+    and :math:`1-p` is the probability of a single failure
+    and :math::`k` is the number of trials to get the first success.
```
Comment (j-bowhay) [inline scipy/special/_add_newdocs.py:13117]:
```suggestion
    and :math:`k` is the number of trials to get the first success.
```

**Review by tupui — APPROVED** [review]
Thank you for the update @OmarManzoor and for using the CI skip comments.

Note that we need to run at least once the Azure ones because the linters are there. We know it's not optimal and are thinking about ways to change this.

Once linters are green LGTM to merge.

Thanks Jake for having a look as well!

**Comment by tupui** [comment]
Actually @OmarManzoor would you mind actually fixing this?

```
scipy/special/_add_newdocs.py:11:10: UP006 [*] Use `dict` instead of `Dict` for type annotations
```

It's not linked to this PR, but does not matter much as it's just a one letter change. It's annoying us because we are touching this file.

**Comment by OmarManzoor** [comment]
> Actually @OmarManzoor would you mind actually fixing this?
> 
> ```
> scipy/special/_add_newdocs.py:11:10: UP006 [*] Use `dict` instead of `Dict` for type annotations
> ```
> 
> It's not linked to this PR, but does not matter much as it's just a one letter change. It's annoying us because we are touching this file.

Sure!

**Comment by tupui** [comment]
Thanks! Please just wait for the `refguide_asv_check` check to finish before pushing with `[skip ci]`.

**Comment by OmarManzoor** [comment]
> Thanks! Please just wait for the `refguide_asv_check` check to finish before pushing with `[skip ci]`.
> 
> Seems I was too late 😅 no worries.

Sorry about that. If needed I'll try to push something after these checks complete.

File: scipy/special/_add_newdocs.py [hunk 2]
```diff
@@ -13103,6 +13102,43 @@ def add_newdoc(name, doc):
...
+    mass function can become very low. In such cases taking the log of the
+    pmf would be more suitable as the log function can change the values
+    to a scale that is more appropriate to work with.
+
+    >>> _log_pmf = xlog1py(k - 1, -p) + np.log(p)
+    >>> _log_pmf
+    -69.31471805599453
+
+    We can confirm that we get a value close to the original pmf value by
+    taking the exponetial of the log pmf.
```
Comment (OmarManzoor) [inline scipy/special/_add_newdocs.py:13136]:
```suggestion
    taking the exponential of the log pmf.
```
This can be fixed.

**Pull request merged**